wandb/docker/__init__.py

```diff
@@ -28,14 +31,91 @@ def shell(cmd):
         return None
 
 
-def default_image(gpu=False):
+_buildx_installed = None
+
+
+def is_buildx_installed() -> bool:
+    """Returns `True` if docker buildx is installed and working."""
+    global _buildx_installed
+    if _buildx_installed is not None:
+        return _buildx_installed
+    help_output = shell(["buildx", "--help"])
+    _buildx_installed = help_output is not None and "buildx" in help_output
+    return _buildx_installed
+
+
+def build(tags: List[str], file: str, context_path: str) -> str:
+    command = ["buildx", "build"] if is_buildx_installed() else ["build"]
+    build_tags = []
+    for tag in tags:
+        build_tags += ["-t", tag]
+    run(
+        ["docker"] + command + build_tags + ["-f", file, context_path],
+        capture_stderr=False,
+        capture_stdout=False,
+    )
+    return tags[0]
+
+
+def run(
+    args: List[Any],
+    capture_stdout: bool = True,
+    capture_stderr: bool = True,
+    input: bytes = None,
+    return_stderr: bool = False,
+    env: Optional[Dict[str, str]] = None,
+) -> Union[str, Tuple[str, str]]:
+    args = [str(x) for x in args]
+    subprocess_env = dict(os.environ)
+    subprocess_env.update(env or {})
+    if args[1] == "buildx":
+        subprocess_env["DOCKER_CLI_EXPERIMENTAL"] = "enabled"
```

Comment:
is this env var checked/used anywhere?

Reply:
It's used by earlier versions of the docker cli.  We need to set it in its environment to use the fancy build caching.